src/target/linux.rs

```diff
@@ -188,4 +193,60 @@ mod tests {
         assert_eq!(read_var2_op, var2);
         assert_eq!(read_var_op, var);
     }
+
+    const PAGE_SIZE: usize = 4096;
+
+    #[test]
+    fn read_protected_memory() {
+        let mut read_var_op: usize = 0;
+
+        unsafe {
+            let ptr = alloc_zeroed(Layout::from_size_align(PAGE_SIZE, PAGE_SIZE).unwrap());
+
+            *(ptr as *mut usize) = 9921;
+
+            mprotect(
+                ptr as *mut std::ffi::c_void,
+                PAGE_SIZE,
+                ProtFlags::PROT_NONE,
+            )
+            .expect("Failed to mprotect");
+
+            let res = ReadMemory::new(getpid())
+                .read(&mut read_var_op, ptr as *const _ as usize)
+                .apply();
+
+            // Expected to fail when reading read-protected memory.
+            // FIXME: Change when reading read-protected memory is handled properly
+            match res {
+                Ok(()) => panic!("Unexpected result: reading protected memory succeeded"),
+                Err(_) => (),
+            }
```

Comment:
`ptr` leaks memory here. We can deallocate it as follows:

```suggestion
            }
            mprotect(
                ptr as *mut std::ffi::c_void,
                PAGE_SIZE,
                ProtFlags::PROT_WRITE | ProtFlags::PROT_READ,
            )
            .expect("Failed to mprotect");
            std::alloc::dealloc(ptr, Layout::from_size_align(PAGE_SIZE, PAGE_SIZE).unwrap());
```